src/clj/witan/send/validate_model.clj

```diff
@@ -0,0 +1,126 @@
+        destination-path (str (temp-dir input-path) year "/")]
+    (.mkdir (java.io.File. destination-path))
+    (doseq [file ["modify-settings.csv" "population.csv" "need-setting-costs.csv" "valid-setting-academic-years.csv"]]
+      (copy-if-exists source-path destination-path file))))
+
+(defn move-target-files [input-path year]
+  (doseq [file ["Output_Count.csv" "Output_AY_State.csv"]]
+    (io/copy (io/file (str "target/" file)) (io/file (str (temp-dir input-path) year "_" file)))))
+
+(defn return-testable-data [file test-years]
+  (->> (for [year test-years] (filter #(= year (:calendar-year %)) (load-csv-as-maps file)))
+       (mapcat seq)))
+
+(defn append-count-with-test [model-count test-data n-transitions]
+  (let [test-year (str (dec (Integer/parseInt (:calendar-year model-count))))
+        test-data-for-year  (filter #(= test-year (:calendar-year %)) test-data)
+        send-test-data-for-year  (filter #(not= "NONSEND" (:need-2 %)) test-data-for-year)
+        ground-truth-count (count send-test-data-for-year)]
+    (assoc model-count :ground-truth (str ground-truth-count) :n-transition n-transitions)))
+
+(defn append-state-with-test [model-state test-data n-transitions]
+  (let [test-year (str (dec (Integer/parseInt (:calendar-year model-state))))
+        state-vector (str/split (str/replace (:state model-state) ":" "") #"-")
+        need (first state-vector)
+        setting (last state-vector)
+        academic-year (:academic-year model-state)
+        test-data-for-state (filter #(and (= test-year (:calendar-year %))
+                                          (= need (:need-2 %))
+                                          (= setting (:setting-2 %))
+                                          (= academic-year (:academic-year-2 %))) test-data)
+        ground-truth-count (count test-data-for-state)]
+    (assoc model-state :ground-truth (str ground-truth-count) :n-transitions (str n-transitions))))
+
+(defn collate-fold [input-path year n-transitions]
+  (let [test-data (load-csv-as-maps (str (temp-dir input-path) "test_" year ".csv"))
+        test-years (->> (map :calendar-year test-data)
+                        (distinct)
+                        (map #(inc (Integer/parseInt %)))
+                        (into [])
+                        (map str))
+        model-state (return-testable-data (str (temp-dir input-path) year "_Output_AY_State.csv") test-years)
+        model-count (return-testable-data (str (temp-dir input-path) year "_Output_Count.csv") test-years)
+        count-results (map #(append-count-with-test % test-data n-transitions) model-count)
+        state-results (map #(append-state-with-test % test-data n-transitions) model-state)]
+    (write-csv (str (temp-dir input-path) "results_" year "_count.csv") count-results)
+    (write-csv (str (temp-dir input-path) "results_" year "_state.csv") state-results)))
+
+(defn validate-fold [input-path year transitions settings]
+  (def temp (temp-dir input-path))
```

Comment:
Why is this `def` and not in the `let` below?

Reply:
good catch, that line shouldn't be there